analyzers/src/SonarAnalyzer.CSharp/Extensions/TypeDeclarationSyntaxExtensions.cs

```diff
@@ -33,14 +33,19 @@ public static class TypeDeclarationSyntaxExtensions
         /// Returns a union of all the methods and local functions from a given type declaration.
         /// </summary>
         public static IEnumerable<IMethodDeclaration> GetMethodDeclarations(this TypeDeclarationSyntax typeDeclaration) =>
-            typeDeclaration
-                .Members
-                .OfType<MethodDeclarationSyntax>()
-                .SelectMany(method => GetLocalFunctions(method).Union(new List<IMethodDeclaration> { MethodDeclarationFactory.Create(method) }));
+            typeDeclaration.Members
+                           .OfType<MethodDeclarationSyntax>()
+                           .SelectMany(method => GetLocalFunctions(method).Union(new List<IMethodDeclaration> { MethodDeclarationFactory.Create(method) }));
 
-        private static IEnumerable<IMethodDeclaration> GetLocalFunctions(MethodDeclarationSyntax methodDeclaration)
-            => methodDeclaration.DescendantNodes()
-                .Where(member => member.IsKind(SyntaxKindEx.LocalFunctionStatement))
-                .Select(member => MethodDeclarationFactory.Create(member));
+        public static IEnumerable<IMethodDeclaration> GetMethodDeclarations(this CompilationUnitSyntax compilationUnitSyntax) =>
+            compilationUnitSyntax.Members.OfType<GlobalStatementSyntax>()
+                                         .Select(x => x.ChildNodes().FirstOrDefault(y => y.IsKind(SyntaxKindEx.LocalFunctionStatement)))
+                                         .Where(x => x != null)
+                                         .Select(x => MethodDeclarationFactory.Create(x));
+
+        private static IEnumerable<IMethodDeclaration> GetLocalFunctions(MethodDeclarationSyntax methodDeclaration) =>
+            methodDeclaration.DescendantNodes()
```

Comment:
This could also capture local functions inside local functions. Is it a desirable behavior?

Reply:
This is not a change for this PR.
 I made the mistake to add the `GetMethodDeclarations(this CompilationUnitSyntax compilationUnitSyntax) ` above and it moved this one lower now it marks it as a change.
Apologies for that!